From 5729a740c1d3b9fb831ea8391310f84dc24b14b2 Mon Sep 17 00:00:00 2001
From: q66 <q66@chimera-linux.org>
Date: Tue, 16 May 2023 22:35:47 +0200
Subject: [PATCH] Make sure days are *really* good enough

This does not fix gnome-calendar in scenarios where /etc/localtime
does not exist, but it at least makes it fail on an assertion
rather than segfaulting on a random buffer overflow, so that
is good enough for now.
---
 src/gui/event-editor/gcal-date-chooser.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/gui/event-editor/gcal-date-chooser.c b/src/gui/event-editor/gcal-date-chooser.c
index 283fe253..58c8744a 100644
--- a/src/gui/event-editor/gcal-date-chooser.c
+++ b/src/gui/event-editor/gcal-date-chooser.c
@@ -756,7 +756,7 @@ multi_choice_changed (GcalDateChooser *self,
   day = g_date_time_get_day_of_month (self->date);
 
   /* Make sure the day is valid at that month */
-  day = MIN (day, month_length[leap (year)][month]);
+  day = MAX (1, MIN (day, month_length[leap (year)][month]));
 
   date = g_date_time_new_local (year, month, day, 0, 0, 0);
   gcal_date_chooser_set_date (GCAL_VIEW (self), date);
-- 
2.39.0

